engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigratorImpl.java

```diff
@@ -120,6 +120,7 @@ private boolean handleMigrationTask(
       return true;
     } else {
       logMigrationSkipped(migrationTask, index, total);
+      markMigrationFinished(migrationTask.getIdentifier());
```

Comment:
🔧 This line means that we **always** mark the migration task as finished, even if it was already marked as finished.

I'd rather change the logic only to mark it when necessary and always look it up. That's important as we're making this new way to decide whether a migration task must still be run. I prefer the following logic:
- check in state if the task has already been finished
- if yes, do nothing
- if no, check if the migration task needs to run (we can provide a default implementation that returns true for all new tasks while keeping the old task check untouched)
- if it needs to run, run it and then mark it as finished
- if it doesn't, still mark it as finished because it wasn't marked yet

Now we only mark it when necessary.